src/test/java/io/aiven/connect/jdbc/sink/BufferedRecordsTest.java

```diff
@@ -164,27 +167,129 @@ public void testInsertModeUpdate() throws SQLException {
 
-        final String url = sqliteHelper.sqliteUri();
-        final DatabaseDialect dbDialect = DatabaseDialects.findBestFor(url, config);
+        final DatabaseDialect dbDialect = DatabaseDialects.findBestFor(dbUrl, config);
         final DbStructure dbStructureMock = mock(DbStructure.class);
-        when(dbStructureMock.createOrAmendIfNecessary(Matchers.any(JdbcSinkConfig.class),
-            Matchers.any(Connection.class),
-            Matchers.any(TableId.class),
-            Matchers.any(FieldsMetadata.class)))
-            .thenReturn(true);
+        when(dbStructureMock.createOrAmendIfNecessary(any(JdbcSinkConfig.class),
+                any(Connection.class),
+                any(TableId.class),
+                any(FieldsMetadata.class)))
+                .thenReturn(true);
 
         final Connection connectionMock = mock(Connection.class);
+        final PreparedStatement preparedStatement = mock(PreparedStatement.class);
+        when(connectionMock.prepareStatement(anyString())).thenReturn(preparedStatement);
+        when(preparedStatement.executeBatch()).thenReturn(new int[1]);
+
         final TableId tableId = new TableId(null, null, "dummy");
         final BufferedRecords buffer = new BufferedRecords(config, tableId, dbDialect, dbStructureMock,
-            connectionMock);
+                connectionMock);
 
         final Schema schemaA = SchemaBuilder.struct().field("name", Schema.STRING_SCHEMA).build();
         final Struct valueA = new Struct(schemaA).put("name", "cuba");
-        final SinkRecord recordA = new SinkRecord("dummy", 0, null, null, schemaA, valueA, 0);
+        final SinkRecord recordA = wrapInSinkRecord(valueA);
         buffer.add(recordA);
+        buffer.flush();
+
+        verify(connectionMock).prepareStatement(eq("UPDATE \"dummy\" SET \"name\" = ?"));
+
+    }
+
+    @Test
+    public void testInsertModeMultiAutomaticFlush() throws SQLException {
+        final JdbcSinkConfig config = multiModeConfig(2);
+
+        final DatabaseDialect dbDialect = DatabaseDialects.findBestFor(dbUrl, config);
+        final DbStructure dbStructureMock = mock(DbStructure.class);
+        when(dbStructureMock.createOrAmendIfNecessary(any(JdbcSinkConfig.class),
+                any(Connection.class),
+                any(TableId.class),
+                any(FieldsMetadata.class)))
+                .thenReturn(true);
+
+        final Connection connection = mock(Connection.class);
+        final PreparedStatement preparedStatement = mock(PreparedStatement.class);
+        when(connection.prepareStatement(anyString())).thenReturn(preparedStatement);
+        when(preparedStatement.executeBatch()).thenReturn(new int[]{2});
+
+        final TableId tableId = new TableId(null, null, "planets");
+        final BufferedRecords buffer = new BufferedRecords(config, tableId, dbDialect, dbStructureMock,
+                connection);
+
+        final Schema schema = newPlanetSchema();
+        for (int i = 1; i <= 5; i++) {
```

Comment:
Is the idea to create five rows, but since the batch size is two, there is not a flush for the last row?

Reply:
That's correct. Basically, a test for the flushing logic too. I'll add a comment, since it looks like it's not that obvious.: )